### PR TITLE
Ensure that we also check for overrides of IDbCommand.CommandText

### LINKED ISSUE #1625: CA2100 does not fire for msdn/docs sample
#### Analyzer package

Example: Microsoft.CodeQuality.Analyzers.Exp, version [2.6.1-beta1-62702-01](https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeQuality.Analyzers.Exp/2.6.1-beta1-62702-01)

#### Analyzer

ReviewSqlQueriesForSecurityVulnerabilities

#### Repro steps

1. Copy the C#or VB sample from https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2100-review-sql-queries-for-security-vulnerabilities

#### Expected behavior
CA2100 fires.

#### Actual behavior
No CA2100

This seems to be due to the fact that `SqlCommand.CommandText` overrides `DbCommand.CommandText`, and the latter implements `IDbCommand.CommandText`, while the analyzer only checks if the invoked property (i.e. `SqlCommand.CommandText`) directly implements `IDbCommand.CommandText`.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #1625

### CODE REVIEW

**Comment by mavasani** [comment]
@333fred @ivanbasov @jinujoseph - please review. Needed to unblock internal dogfooding for this rule.

File: src/Microsoft.CodeQuality.Analyzers.Exp/Core/Security/ReviewSqlQueriesForSecurityVulnerabilities.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public override void Initialize(AnalysisContext context)
                         var propertyReference = (IPropertyReferenceOperation)operationContext.Operation;
 
                         // We're only interested in implementations of IDbCommand.CommandText
-                        if (!propertyReference.Property.IsImplementationOfInterfaceMember(commandTextProperty))
+                        if (!propertyReference.Property.IsOverrideOrImplementationOfInterfaceMember(commandTextProperty))
```
Comment (mavasani) [inline src/Microsoft.CodeQuality.Analyzers.Exp/Core/Security/ReviewSqlQueriesForSecurityVulnerabilities.cs]: Filed https://github.com/dotnet/roslyn-analyzers/issues/1627 to audit other usages of `IsImplementationOfInterfaceMember`.

**Comment by mavasani** [comment]
@dotnet-bot retest license/cla please

**Comment by 333fred** [comment]
LGTM

**Pull request merged**